### PR TITLE
COZMO-10302 Custom object markers

### PR DESCRIPTION
**Pull request by MarkWesleyAtAnki**
Added support for custom cubes, boxes and walls using custom markers. Can be marked as unique (only 1 of that type, with those markers, in the world) with is_unique flag to allow the engine to assume all instances of the marker are the same object ID.
Added / exposed CustomObjectTypes and CustomObjectMarkers
Increased OBJECT_VISIBILITY_TIMEOUT from 0.2 to 0.4 seconds - during slow updates it’s possible to miss an observed message for over 0.2 seconds
Added example that creates multiple custom objects and logs when they become visible or go out of view.
Add all the markers to the repo (in the docs to make it easy to document them), and removed the 2 old markers

### CODE REVIEW

File: examples/tutorials/04_cubes_and_objects/09_custom_objects.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+'''
+
+import time
+
+import cozmo
+from cozmo.objects import CustomObject, CustomObjectMarkers, CustomObjectTypes
+
+
+def handle_object_appeared(evt, **kw):
+    # This will be called whenever an EvtObjectAppeared is dispatched - so
```
Comment (msintov) [inline examples/tutorials/04_cubes_and_objects/09_custom_objects.py]: I would remove the 'so' here can just leave it as "whenever an Object comes into view."

File: examples/tutorials/04_cubes_and_objects/09_custom_objects.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    # whenever an Object comes into view.
+    if isinstance(evt.obj, CustomObject):
+        print("Cozmo started seeing a %s" % str(evt.obj.object_type))
+
+
+def handle_object_disappeared(evt, **kw):
+    # This will be called whenever an EvtObjectDisappeared is dispatched - so
```
Comment (msintov) [inline examples/tutorials/04_cubes_and_objects/09_custom_objects.py]: Again I'd remove the "so".

File: examples/tutorials/04_cubes_and_objects/09_custom_objects.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                                              30, 30, True)
+
+    # define a unique cube (88mm x 88mm x 88mm) (approximately 2x the size of a light cube)
+    # with a 50mm x 50mm Diamonds3 image on every face
+    big_cube_obj = robot.world.define_custom_cube(CustomObjectTypes.CustomType01,
+                                              CustomObjectMarkers.Diamonds3,
+                                              88,
+                                              50, 50, True)
+
+    # define a unique wall (150mm x 120mm (x10mm thick for all walls)
```
Comment (msintov) [inline examples/tutorials/04_cubes_and_objects/09_custom_objects.py]: Where is the 10mm thickness specified?

File: examples/tutorials/04_cubes_and_objects/09_custom_objects.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                                            30, 50, True)
+
+    if ((cube_obj is not None) and (big_cube_obj is not None) and
+            (wall_obj is not None) and (box_obj is not None)):
+        print("All objects defined successfully!")
+    else:
+        print("One or more object definitions failed!")
+        return
+
+    print("Show the above icons to Cozmo and you will see the related objects "
```
Comment (msintov) [inline examples/tutorials/04_cubes_and_objects/09_custom_objects.py]: When this prints out, what exactly is the text referring to when it says "Show the above icons to Cozmo?" Is there some icon representation in the output?

File: src/cozmo/objects.py [hunk 5]
```diff
@@ -41,7 +41,7 @@
            'EvtObjectAppeared', 'EvtObjectAvailable', 'EvtObjectTapped',
            'EvtObjectConnectChanged', 'EvtObjectDisappeared', 'EvtObjectObserved',
            'ObservableElement', 'ObservableObject', 'LightCube', 'Charger',
-           'CustomObject', 'FixedCustomObject']
+           'CustomObject', 'CustomObjectTypes', 'CustomObjectMarkers', 'FixedCustomObject']
```
Comment (msintov) [inline src/cozmo/objects.py]: Should these two additions be in alpha order?

File: src/cozmo/objects.py [hunk 6]
```diff
@@ -537,19 +542,145 @@ def marker_height_mm(self):
...
+    For use with world.define_custom methods such as
+    :meth:`cozmo.world.World.define_custom_box`,
+    :meth:`cozmo.world.World.define_custom_cube`, and
+    :meth:`cozmo.world.World.define_custom_wall`
+    '''
+
+    #: CustomType00 - the first custom object type
+    CustomType00 = _CustomObjectType("CustomType00", _clad_to_engine_cozmo.ObjectType.CustomType00)
+
+    #:
```
Comment (msintov) [inline src/cozmo/objects.py]: Why are there "#:" throughout here?

File: src/cozmo/world.py [hunk 7]
```diff
@@ -590,73 +590,207 @@ def send_available_objects(self):
...
+                affixed to the back of the object
+            marker_top (:class:`cozmo.objects.CustomObjectMarkers`): the marker
+                affixed to the top of the object
+            marker_bottom (:class:`cozmo.objects.CustomObjectMarkers`): the marker
+                affixed to the bottom of the object
+            marker_left (:class:`cozmo.objects.CustomObjectMarkers`): the marker
+                affixed to the left of the object
+            marker_right (:class:`cozmo.objects.CustomObjectMarkers`): the marker
+                affixed to the right of the object
+            depth_mm (float): depth of the object (in millimeters) (X axis).
```
Comment (msintov) [inline src/cozmo/world.py]: In the depth_mm, width_mm and height_mm lines, I don't think we need a period after the "(* axis)." If we have a period after "(in millimeters)" it should uniform across the 3 lines.

File: src/cozmo/world.py [hunk 8]
```diff
@@ -590,73 +590,207 @@ def send_available_objects(self):
...
+            width_mm (float): width of the object (in millimeters). (Y axis).
+            height_mm (float): height of the object (in millimeters). (Z axis).
+                (the height of the object)
+            marker_width_mm (float): width of the printed marker (in millimeters).
+            maker_height_mm (float): height of the printed marker (in millimeters).
+            is_unique (bool): can the engine assume there is only 1 of this object
```
Comment (msintov) [inline src/cozmo/world.py]: I would write the docs for is_unique to: "If true, the engine will assume there is only 1 etc.".

File: src/cozmo/world.py [hunk 9]
```diff
@@ -590,73 +590,207 @@ def send_available_objects(self):
...
-            y_size_mm (float): size of the object (in millimeters) in the y axis.
-            z_size_mm (float): size of the object (in millimeters) in the z axis.
+            custom_object_type (:class:`cozmo.objects.CustomObjectTypes`): the
+                object type you are binding this custom object to.
+            marker:(:class:`cozmo.objects.CustomObjectMarkers`): the marker
+                affixed to every side of the cube.
+            size_mm: size of each side of the cube (in millimeters).
             marker_width_mm (float): width of the printed marker (in millimeters).
             maker_height_mm (float): height of the printed marker (in millimeters).
+            is_unique (bool): can the engine assume there is only 1 of this object
```
Comment (msintov) [inline src/cozmo/world.py]: Same is_unique doc comment as above.

File: src/cozmo/world.py [hunk 10]
```diff
@@ -590,73 +590,207 @@ def send_available_objects(self):
...
+                object type you are binding this custom object to.
+            marker:(:class:`cozmo.objects.CustomObjectMarkers`): the marker
+                affixed to the front and back of the wall
+            width_mm (float): width of the object (in millimeters). (Y axis).
+            height_mm (float): height of the object (in millimeters). (Z axis).
+            width_mm: width of the wall (along Y axis) (in millimeters).
+            height_mm: height of the wall (along Z axis) (in millimeters).
+            marker_width_mm (float): width of the printed marker (in millimeters).
+            maker_height_mm (float): height of the printed marker (in millimeters).
+            is_unique (bool): can the engine assume there is only 1 of this object
```
Comment (msintov) [inline src/cozmo/world.py]: Same is_unique doc comment as above.

File: examples/tutorials/04_cubes_and_objects/09_custom_objects.py (hunk 4 above)
Comment (MarkWesleyAtAnki) [inline examples/tutorials/04_cubes_and_objects/09_custom_objects.py]: I'll reword it to markers. You show Cozmo the marker (e.g. CustomObjectMarkers.Diamonds2) and he'll observe that object (CustomObjectTypes.CustomType00)

File: examples/tutorials/04_cubes_and_objects/09_custom_objects.py (hunk 3 above)
Comment (MarkWesleyAtAnki) [inline examples/tutorials/04_cubes_and_objects/09_custom_objects.py]: 10mm is hardcode in engine, there's a comment in world.py related to that

File: src/cozmo/objects.py (hunk 6 above)
Comment (MarkWesleyAtAnki) [inline src/cozmo/objects.py]: With a `#:` in front of it they're listed in the documentation, otherwise there's no way to see them. I excluded a text description for each (just one for the 1st and last) as that kept that generated documentation looking cleaner.
